src/goto-synthesizer/cegis_verifier.h

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************\
+
+Module: Verifier for Counterexample-Guided Synthesis
+
+Author: Qinheping Hu
+
+\*******************************************************************/
+
+/// \file
+/// Verifier for Counterexample-Guided Synthesis
+
+#ifndef CPROVER_GOTO_SYNTHESIZER_CEGIS_VERIFIER_H
+#define CPROVER_GOTO_SYNTHESIZER_CEGIS_VERIFIER_H
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/loop_ids.h>
+
+#include <goto-checker/all_properties_verifier.h>
+
+#include "synthesizer_utils.h"
+
+class messaget;
+
+/// Formatted counterexample.
+class cext
+{
+public:
+  enum class violation_typet
+  {
+    cex_out_of_boundary,
+    cex_null_pointer,
+    cex_not_preserved,
+    cex_not_hold_upon_entry,
+    cex_other
+  };
+
+  cext(
+    const std::map<exprt, std::size_t> &object_sizes,
+    const std::map<exprt, std::size_t> &havoced_values,
+    const std::map<exprt, std::size_t> &havoced_pointer_offsets,
+    const std::map<exprt, std::size_t> &loop_entry_values,
+    const std::map<exprt, std::size_t> &loop_entry_offsets,
+    const std::set<exprt> &live_variables)
+    : object_sizes(object_sizes),
+      havoced_values(havoced_values),
+      havoced_pointer_offsets(havoced_pointer_offsets),
+      loop_entry_values(loop_entry_values),
+      loop_entry_offsets(loop_entry_offsets),
+      live_variables(live_variables)
+  {
+  }
+
+  cext()
+  {
+  }
```

Comment:
Safe to omit

Reply:
Seems this still needs to be removed?

Reply:
I tried to omit it. But failed to compile with error 
```
/Users/qinhh/Repos/CBMC/clone_2/src/goto-synthesizer/cegis_verifier.h:90:3: error: constructor for 'cegis_verifiert' must explicitly initialize the member 'return_cex' which does not have a default constructor
  cegis_verifiert(
  ^
/Users/qinhh/Repos/CBMC/clone_2/src/goto-synthesizer/cegis_verifier.h:105:8: note: member is declared here
  cext return_cex;
       ^
/Users/qinhh/Repos/CBMC/clone_2/src/goto-synthesizer/cegis_verifier.h:25:7: note: 'cext' declared here
class cext
      ^
1 error generated.
```

Reply:
Perhaps the use should be fixed? A default-constructed `cext` seems ill-formed. Perhaps that `return_cex` class member shouldn't exist?

Reply:
Yes, the return value of `verify()` is now `optionalt<cext>`---it return a counterexample if there is some violation.